DesignAutomationHandler2018/DesignAutomationHandler.cs

```diff
@@ -13,6 +16,36 @@ class DesignAutomationHandlerApp : IExternalCommand
         public Result Execute(ExternalCommandData commandData, ref string message, ElementSet elements)
         {
             var app = commandData.Application.Application;
+            string[] files = Directory.GetFiles(app.AllUsersAddinsLocation, "*.addin");
+            foreach (string file in files)
+            {
+               XElement addin = XElement.Load(file);
+               IEnumerable<XElement> childList = from el in addin.Elements() select el;
+               foreach (XElement e in childList)
+               {
+                  try
+                  {
+                     System.Reflection.Assembly a = System.Reflection.Assembly.LoadFile(e.Element("Assembly").Value);
+                     bool designAutomationBridge = false;
+                     bool revitAPIUI = false;
+                     foreach (System.Reflection.AssemblyName an in a.GetReferencedAssemblies())
+                     {
+                        string assemblyName = an.Name;
+                        if (assemblyName == "DesignAutomationBridge")
+                           designAutomationBridge = true;
+                        else if (assemblyName == "RevitAPIUI")
+                           revitAPIUI = true;
+                     }
```

Comment:
I think there is easier way to tell if it is a db level dll or ui level addin by check
> <AddIn Type="DBApplication">

Reply:
I hope so; I don't like this way of checking very much. Do you think there is a better option?